packages/shorebird_cli/lib/src/android_sdk.dart

```diff
@@ -0,0 +1,55 @@
+import 'dart:io';
+
+import 'package:collection/collection.dart';
+import 'package:path/path.dart' as p;
+import 'package:scoped/scoped.dart';
+import 'package:shorebird_cli/src/platform.dart';
+
+// https://developer.android.com/studio/command-line/variables.html#envar
+const kAndroidHome = 'ANDROID_HOME';
+const kAndroidSdkRoot = 'ANDROID_SDK_ROOT';
+
+/// A reference to a [AndroidSdk] instance.
+final androidSdkRef = create(AndroidSdk.new);
+
+/// The [AndroidSdk] instance available in the current zone.
+AndroidSdk get androidSdk => read(androidSdkRef);
+
+class AndroidSdk {
+  String? get path {
+    String? androidHomeDir;
+    if (platform.environment.containsKey(kAndroidHome)) {
```

Comment:
For consistency with the other android-related files, we should probably either stick to a cascading if/else (as here) or early returns (as elsewhere)